### DISCUSSION TITLE
Working Watermeters

### DISCUSSION THREAD

**Discussion by huizebruin** [huizebruin/s0tool#57] [General]
At this moment i know this ones are working.

Watermerk type :	Extra info :	Land:	Sensor
Itron		NL	Npn sensor
Sensus		NL	Npn sensor # Metaal zit in het rode wijzertje
Actaris		NL	Npn sensor
Elster		NL	Npn sensor
Maddalena	BE	Npn sensor

or other meters but the must have a turning weel with metal inside or on top of it.

![watermeter](https://user-images.githubusercontent.com/62996429/199828840-28d2cf97-9a30-4504-9ce9-1d596ab04272.jpg)

When u have a other meter u can post it here.

**Comment by DJBenson** [comment]
I think this Elster meter (UK) will work?

![IMG_7412](https://github.com/huizebruin/s0tool/assets/1013909/64e1824a-146a-4734-9adb-71439ece3c34)

**Reply by huizebruin** [reply]
i see a metal half moon on the right bottom, then it will work with a 8mm npn sensor.

**Comment by cleveHEX** [comment]
![image](https://github.com/huizebruin/s0tool/assets/9021297/9a6a888d-c5e0-4b4c-ac6d-547ae2285250)
I guess this could work too?

**Reply by huizebruin** [reply]
Yes I think it's going to work on your meter.

**Comment by physje** [comment]
This works with me.
![ElsterV200](https://github.com/huizebruin/s0tool/assets/3999563/b35f681c-bc5f-467b-b4e2-cbed18d39bdb)

Elster V200 in combination with a TL-W5MC1 proximity sensor (C1-5V NPN NO; [AliExpress](https://nl.aliexpress.com/item/32582384960.html)).
The bracket is from [Willem Hol](https://www.thingiverse.com/willemah) on [Thingiverse ](https://www.thingiverse.com/thing:6088225)

**Reply by physje** [reply]
Don't know if it's due to the sensor or something else, but the watermeter is not very stable lately.
It hangs quite a lot.
Curious to see if more people have the same experience with this sensor

**Reply by huizebruin** [reply]
What is your problem?

I'm using it now for 2 years without any problem.

Did you build the S0tool by yourself?

If you're interested and you're a Dutch or Belgium person, check out my discord, link is on my website.
And we can chat about your problems in Dutch. There are a lot more users in the group that are using it for a while.

**Comment by HobbitBluelabel** [comment]
Hi,

In my case the watermeter is very stable for over a year now, no problems
at all...

Hartelijke groet,
Jan-Willem van den Hoek


Op wo 6 dec 2023 om 18:31 schreef Matthijs Draijer ***@***.***

**Reply by physje** [reply]
The Elster V200 in combination with a TL-W5MC1 proximity sensor is stable in you case?
That means, I cannot blame the sensor.

**Comment by HobbitBluelabel** [comment]
Maybe it is better to define "unstable"
In your case. What is happening, how frequent and how do you solve it?
What did you try to solve it?
What hardware/software do you use?
A.s.o.

Op wo 6 dec 2023 18:36 schreef Matthijs Draijer ***@***.***>:

**Comment by teras** [comment]
Hello community. I am thinking of making my own meter as well, since I need a ESP32 (I'll try to make it also a BT hotspot, so that I'll be informed with a Mi Flora device the state of the soil) and not a ESP8266 that the ready kits provide).

I am confused though if the [`C1-5V NPN NO` device ](https://nl.aliexpress.com/item/32582384960.html?gatewayAdapt=glo2nld), as mentioned [earlier](https://github.com/huizebruin/s0tool/discussions/57#discussioncomment-7473173), would work with confidence. I prefer the flat setting , since my meter is on the wall, and it'll be a PITA to make it with the regular sensor, @HobbitBluelabel do you have the same flat sensor? Should I go with this design or should I do something different, based on @physje feedback?

**Reply by huizebruin** [reply]
Esp32 doesn't handle 5v to the output, and ports. 
The only one is at this point the esp82 Wemos d1 mini.

**Reply by teras** [reply]
This makes sense. One more question. Here the meters that are sold don't have this metallic "moon like" pointer. Still, all of them claim that there are using magnetic transmission ; but probably this is not enough. Should this "moon-like" pointer exist, or, as far as you know, other types might work too?

**Reply by huizebruin** [reply]
Do you have a picture of it?

**Reply by teras** [reply]
I haven't bought it yet; still evaluating my options.
<details>
  <summary>In order to reduce clutter in this thread, I put the images here. Please click to open it</summary>

![image](https://github.com/huizebruin/s0tool/assets/592568/01041b41-3e5e-4ec3-921d-7b58f792f8cc)

![image](https://github.com/huizebruin/s0tool/assets/592568/a450f532-f8f4-457f-a5fa-3d4c31154a4b)

![image](https://github.com/huizebruin/s0tool/assets/592568/f884fdce-9ffc-44c4-8e5d-e49d9b85e15a)

</details>

**Comment by Atlantis1971** [comment]
Hi there,

I have this water counter. I suppose it's not right to realize the water meter as described above, isn't it?

![2019-08-01 - Progetto P1](https://github.com/huizebruin/s0tool/assets/77482474/bf2e20e5-8acf-40bb-b80a-5a03f8c35890)

If not, could you tell me which products are better to buy between Itron, Sensus, Actaris, Elster and Maddalena ones?

Thank you.

Alex

**Reply by huizebruin** [reply]
I dont know that type of meter.
i dont know witch type is better , i have here the Sensus meter.

**Comment by Atlantis1971** [comment]
Hi there,
thank you to @huizebruin for the answer.

So, considering I will change it, anyone knows a water meter running in Home Assistant?

Thank you.

Alex

**Comment by gabter1** [comment]
My Maddalena CD SD Plus water meter is listed on the front page as working with s0tool. I created one DIY, the sensor is tested working, however, I cannot get a reading. Can @huizebruin or someone please point me to where I should place my proximity sensor for this to work? I can't see anything made of metal on this meter. This is what it looks like: 
![IMG20260331073803](https://github.com/user-attachments/assets/470e91aa-a3d7-4489-aad9-8335fc6816b8)
Thanks!

**Reply by huizebruin** [reply]
<img width="141" height="129" alt="image" src="https://github.com/user-attachments/assets/7a9c1fe7-35c9-40fc-9370-8332e34ad10a" />

Did you try it on this one? You should place the sensor on the edge of the rotating wheel (about halfway across), where the contrast or movement is easiest to detect.

**Reply by gabter1** [reply]
Tanks, I tried that as well and couldn't get a reading, unfortunately. I am starting to believe that the meter given by the utility company may not have a metal part at all. I can somwhat see inside the meter from the side and I only see plastic cogwheels inside.

**Reply by huizebruin** [reply]
Then i will remove the **Maddalena CD SD Plus**  from the list for now.
Thanks for testing.
And giving feedback
